js/ui/tab_cookies.js

```diff
@@ -156,6 +156,52 @@
           }
         });
         break;
+      case "whitelist-cookie-domain":
+        var domain = getParentData(element, "data-access");
+        var url = getUrl(domain, "", true);
+        getStorage("cookieWhitelist", function(data) 
+        {
+          let whitelist = data["cookieWhitelist"]
+          if (!whitelist.propertyIsEnumerable(domain)) 
+          {
+            whitelist[domain] = []
+          }
+          if (!whitelist[domain].includes("")) 
+          {
+            data["cookieWhitelist"][domain].push("")
+            setStorage(data)
+          } 
+          else 
+          {
+            whitelist[domain] = whitelist[domain].filter(el => el !== "")
+            setStorage(data)
+          }
+        });
+        break;
+      case "whitelist-sublist-cookie":
+        var accessObj = JSON.parse(getParentData(element, "data-access"));
+        var domain = getParentData(getParentData(element, "data-access", true).
+          parentElement, "data-access");
+        var url = getUrl(domain, accessObj.path, accessObj.secure);
+        getStorage("cookieWhitelist", function(data) 
+        {
+          let whitelist = data["cookieWhitelist"]
+          if (!whitelist.propertyIsEnumerable(domain)) 
+          {
+            whitelist[domain] = []
+          }
+          if (!whitelist[domain].includes(accessObj.cookie)) 
+          {
+            data["cookieWhitelist"][domain].push(accessObj.cookie)
```

Comment:
Detail: You can reference here to `whitelist `, same as you have done below:
```
whitelist[domain]...
```

Also `setStorage()` can be moved out of the `if-else` cases to avoid duplications: